crates/flox-rust-sdk/src/models/environment.rs

```diff
@@ -536,6 +590,91 @@ async fn cp_r(from: impl AsRef<Path>, to: &impl AsRef<Path>) -> Result<(), std::
     Ok(())
 }
 
+/// insert packages into the content of a flox.nix file
+fn flox_nix_content_with_new_packages(
+    flox_nix_content: &str,
```

Comment:
```suggestion
    flox_nix_content: impl AsRef<str>,
```

This is more flexible so I would go with this unless it causes major complications